First draft for InFocus support.

petterreinholdtsen opened the pull request:
This patch is untested, but coded based on a standalone client I have been using.  Is there a way to
test this outside kodi?

fredrik-eriksson commented:
Nice! When I tested this previously I think I created a test script that included the projector class and ran a few manual commands to make sure the projector communication code worked. Not very scientific, but it worked well enough. But the easiest way by far is to simply install it in kodi and make sure everything is working. You can even check out the repository directly to your kodi addon directory (as service.projcontrol) and work with it directly from there.

I think there is an error in your entry for infocus_model in settings.xml. visible="eq(-1,InFocus)" should probably be visible="eq(-2,InFocus)", because it's the entry two steps below you want to test against. It  has been a long time since I looked at the syntax, there may be better ways to test against other entries, but that was what I came up with at the time...

petterreinholdtsen commented:
OK.  Changed the index and rebased the branch based on your updated master.  I tested the code using some dummy classes, available from my standalone-testing branch.  Not sure if is fit for inclusion in the official version.

fredrik-eriksson commented:
Well, I don't have an InFocus projector, so I'll take your word for that it's working, that's good enough for me :)
